Fix some log levels and reduce (debug) log spam

uklotzde opened the pull request:
While testing I noticed that we still write too many debug logs that should only be written while debug logging is enabled explicitly.

I've divided the modifications into commits with appropriate commit messages for easy verification.

uklotzde committed:
Raise some log levels from debug to info/warning

uklotzde committed:
Lower log level for non-critical MP3 decoding errors

uklotzde committed:
Only log debug messages if debug logging is enabled

daschuer commented on src/sources/metadatasourcetaglib.cpp:
```diff
@@ -745,9 +749,11 @@ class SafelyWritableFile final {
-            << "into file" << m_fileName
-            << "with type" << m_fileType;
+    if (kLogger.debugEnabled()) {
+        kLogger.debug() << "Exporting track metadata"
```
I am not sure if the file name is logged somewhere else as well. If not this one is important to identify a faulty file that might be the cause for a taglib crash.

daschuer commented on src/sources/soundsourceproxy.cpp:
```diff
@@ -618,12 +620,14 @@ mixxx::AudioSourcePointer SoundSourceProxy::openAudioSource(const mixxx::AudioSo
-                << openMode;
+        if (kLogger.debugEnabled()) {
+            kLogger.debug() << "Opening file"
+                    << getUrl().toString()
```
This one can be also a helper for identifying a crashing file.

daschuer commented:
I think we should keep at least the file related messages to identify a faulty file.

uklotzde committed:
Re-enable file related logs

uklotzde commented:
Ok. I've re-enabled the two file related logs and also added comments why we decided do this.

daschuer commented:
Thank you. Unfortunately Mac build fails 

```
Undefined symbols for architecture x86_64:
  "mixxx::Logging::s_logLevel", referenced from:
      mixxx::Logging::enabled(mixxx::LogLevel) in metadatasourcetaglib.os
      mixxx::Logging::enabled(mixxx::LogLevel) in trackmetadatataglib.os
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [osx64_build/plugins/osx64_build/m4a/libsoundsourcem4a.dylib] Error 1
scons: building terminated because of errors.
```

daschuer commented:
Do you have a idea to fix the builds soon, or should we remove this from the 2.1.1 milestone?

uklotzde committed:
Fix propagation of log levels into plugins with an ugly hack

The configured log levels have to be propagated into the plugins
upon initialization! This was already broken before I submitted
the previous changes that finally broke the plugin build on macOS.

uklotzde committed:
Merge branch '2.1' into 2.1_reduce_log_spam

daschuer commented:
LGTM, Is this still [WIP]?

uklotzde commented:
No, I just wanted to await the build results. Building locally with clang on Linux worked, I was not able to reproduce the linker error.

Unfortunately I had to bump the API version. This is a really ugly hack and only a partial fix. But we can't fix the whole logging with its global state variables that don't get initialized properly in the plugins. We really need to get rid of the plugins.

daschuer commented:
OK, the CI fails unrelated. Merge?

uklotzde commented:
There is still a small chance that the Windows build might fail. Although according to my understanding this new version should at least be _less broken_ than before ;)

uklotzde commented:
I need to test it again. Currently the plugins fail to load even on Linux!?!

ninomp commented on src/sources/soundsourcepluginapi.h:
```diff
@@ -3,6 +3,7 @@
 
 #define MIXXX_SOUNDSOURCEPLUGINAPI_VERSION 17
```
Shouldn't this be 18?

uklotzde committed:
Bump API version (now actually)

uklotzde commented:
@ninomp Thanks for the hint. Not the first time that this happened.

daschuer commented:
Ok then let's merge it, and see what happens.

daschuer commented:
This was one of the last 2.1.1 PRs.
Would you mind to look into: https://github.com/mixxxdj/mixxx/pull/1691?
